src/app/shared/form/form.service.ts

```diff
@@ -105,6 +108,32 @@ export class FormService {
     });
   }
 
+  public hasValidationErrors(formGroup: FormGroup | FormArray): boolean {
+    let hasErrors = false;
+    const fields: string[] = Object.keys(formGroup.controls);
+    for (const field of fields) {
+    // Object.keys(formGroup.controls).forEach((field) => {
+      const control = formGroup.get(field);
+      if (control instanceof FormControl) {
+        hasErrors = !control.valid && control.touched;
+      } else if (control instanceof FormGroup || control instanceof FormArray) {
+        hasErrors = this.hasValidationErrors(control);
+      }
+      if (hasErrors) {
+        break;
+      }
+    // });
```

Comment:
Same here, comment can be removed it seems

Reply:
done